src/main/java/org/example/app/service/TwitService.java

```diff
@@ -0,0 +1,67 @@
+package org.example.app.service;
+
+import lombok.AllArgsConstructor;
+import lombok.SneakyThrows;
+import org.example.app.solrclient.TwitSolrClient;
+import org.example.app.entity.Twit;
+import org.example.app.repository.TwitRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Service
+@AllArgsConstructor
+public class TwitService {
+
+    @Autowired
+    private final TwitRepository twitRepository;
+
+    private final TwitSolrClient solrClient = new TwitSolrClient();
```

Comment:
We can declare TwitSolrClient as a spring component as well and inject it to this class

Reply:
fixed